### PR TITLE
documentation richard

### PR DESCRIPTION
**Pull request by r-ismer**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/blindly/location/AndroidLocationService.kt [hunk 1]
```diff
@@ -21,36 +24,55 @@ class AndroidLocationService(private var context: Context) : LocationService {
...
+                        LocationManager.GPS_PROVIDER,
+                        MIN_TIME_FOR_UPDATE,
+                        MIN_DISTANCE_FOR_UPDTAE,
+                        this
+                    )
                     locationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER)
                 } else {
-                    //get location from Network
-                    locationManager.requestLocationUpdates(LocationManager.NETWORK_PROVIDER, MIN_TIME_FOR_UPDATE, MIN_DISTANCE_FOR_UPDTAE, this)
+                    //get location form network
```
Comment (r-ismer) [inline app/src/main/java/ch/epfl/sdp/blindly/location/AndroidLocationService.kt:55]:
```suggestion
                    //get location from network
```

File: app/src/main/java/ch/epfl/sdp/blindly/location/AndroidLocationService.kt [hunk 2]
```diff
@@ -21,36 +24,55 @@ class AndroidLocationService(private var context: Context) : LocationService {
...
         } catch (e: SecurityException) {
             throw e
         }
         return location
     }
 
+    /**
+     * Change the user's location if the previous one is different from the actual one
+     *
+     * @param loc the users previous location
```
Comment (r-ismer) [inline app/src/main/java/ch/epfl/sdp/blindly/location/AndroidLocationService.kt:74]:
```suggestion
     * @param loc the user's previous location
```

File: app/src/main/java/ch/epfl/sdp/blindly/match/CardStackAdapter.kt [hunk 3]
```diff
@@ -8,15 +8,33 @@ import android.widget.TextView
...
+ * An adapter to display the cards in the match activity as a stack
+ *
+ * @property profiles the profiles to show
+ */
 class CardStackAdapter(
     private var profiles: List<Profile> = emptyList()
 ) : RecyclerView.Adapter<CardStackAdapter.ViewHolder>() {
 
+    /**
+     * Called when the RecyclerViewnedds a new ViewHolder of the given type to represent an item
```
Comment (r-ismer) [inline app/src/main/java/ch/epfl/sdp/blindly/match/CardStackAdapter.kt:21]:
```suggestion
     * Called when the RecyclerView needs a new ViewHolder of the given type to represent an item
```

File: app/src/main/java/ch/epfl/sdp/blindly/match/CardStackAdapter.kt [hunk 4]
```diff
@@ -8,15 +8,33 @@ import android.widget.TextView
...
+     * @param viewType the view type of the new View
+     * @return a ViewHolder that holds a View of the given type
+     */
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
         val inflater = LayoutInflater.from(parent.context)
         return ViewHolder(inflater.inflate(R.layout.item_profile, parent, false))
     }
 
+    /**
+     * Called by RecyclerView to display the data at the precified position
```
Comment (r-ismer) [inline app/src/main/java/ch/epfl/sdp/blindly/match/CardStackAdapter.kt:33]:
```suggestion
     * Called by RecyclerView to display the data at the specified position
```

**Review by cmsigrist — APPROVED** [review]
Looks all good to me

File: app/src/main/java/ch/epfl/sdp/blindly/location/AndroidLocationService.kt [hunk 5]
```diff
@@ -21,38 +24,57 @@ class AndroidLocationService(private var context: Context) : LocationService {
...
+     *
+     * @return the users location or null if GPS and network are disable
+     */
     override fun getCurrentLocation(): Location? {
         try {
+            //Get the location manager
             locationManager = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
 
-            //check for GPS
+            //Check if the GPS is enable
```
Comment (cgntnr) [inline app/src/main/java/ch/epfl/sdp/blindly/location/AndroidLocationService.kt:38]: Small typo here, should be "is enabled".

File: app/src/main/java/ch/epfl/sdp/blindly/location/AndroidLocationService.kt [hunk 6]
```diff
@@ -21,38 +24,57 @@ class AndroidLocationService(private var context: Context) : LocationService {
...
             isGPSEnable = locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)
 
-            //check for network
+            //Check if the network is enable
```
Comment (cgntnr) [inline app/src/main/java/ch/epfl/sdp/blindly/location/AndroidLocationService.kt:41]: Same goes for here as well "enabled".

File: app/src/main/java/ch/epfl/sdp/blindly/match/MatchActivity.kt [hunk 7]
```diff
@@ -69,37 +78,57 @@ class MatchActivity : AppCompatActivity(), CardStackListener {
     override fun onCardCanceled() {
     }
 
+    /**
+     * Do some action when the card appear
```
Comment (cgntnr) [inline app/src/main/java/ch/epfl/sdp/blindly/match/MatchActivity.kt:82]: appears*

File: app/src/main/java/ch/epfl/sdp/blindly/match/MatchActivity.kt [hunk 8]
```diff
@@ -69,37 +78,57 @@ class MatchActivity : AppCompatActivity(), CardStackListener {
...
+     *
+     * @param view in which the card is
+     * @param position in the view
+     */
     override fun onCardAppeared(view: View, position: Int) {
     }
 
+    /**
+     * Do some action when the card disappear
```
Comment (cgntnr) [inline app/src/main/java/ch/epfl/sdp/blindly/match/MatchActivity.kt:91]: disappears*

File: app/src/main/java/ch/epfl/sdp/blindly/match/Profile.kt [hunk 9]
```diff
@@ -37,7 +37,7 @@ data class Profile(
     /**
      * Describe the content as in Integer
      *
-     * @return content description
+     * @return content description (0)
```
Comment (cgntnr) [inline app/src/main/java/ch/epfl/sdp/blindly/match/Profile.kt:40]: I didn't understand what this is doing only returning 0, maybe explain a bit more in the comments why this is needed.

**Review by cgntnr — APPROVED** [review]

File: app/src/main/java/ch/epfl/sdp/blindly/LocationPermissionActivity.kt [hunk 10]
```diff
@@ -9,6 +9,10 @@ import androidx.appcompat.app.AppCompatActivity
 import androidx.core.app.ActivityCompat
 import ch.epfl.sdp.blindly.profile_setup.ProfileName
 
+/**
+ * Activity that require the user to enable location service for the app to use it
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/LocationPermissionActivity.kt:13]:
```suggestion
 * Activity that requires the user to enable location service for the app to use it
```

File: app/src/main/java/ch/epfl/sdp/blindly/LocationPermissionActivity.kt [hunk 11]
```diff
@@ -19,14 +23,26 @@ class LocationPermissionActivity : AppCompatActivity() {
 
         val button: Button = findViewById(R.id.button)
 
+        /**
+         * When clicking on the button require the location service to be able
+         * to continue with the profile setup
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/LocationPermissionActivity.kt:28]:
```suggestion
         * Clicking on the button requires the location service to be able
         * to continue with the profile setup
```

File: app/src/main/java/ch/epfl/sdp/blindly/location/AndroidLocationService.kt (hunk 5 above)
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/location/AndroidLocationService.kt:38]:
```suggestion
            //Check if the GPS is enabled
```

File: app/src/main/java/ch/epfl/sdp/blindly/location/AndroidLocationService.kt (hunk 6 above)
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/location/AndroidLocationService.kt:41]:
```suggestion
            //Check if the network is enabled
```

File: app/src/main/java/ch/epfl/sdp/blindly/match/MatchActivity.kt [hunk 12]
```diff
@@ -20,6 +20,9 @@ private const val SCALE_INTERVAL = 0.95f
 private const val SWIPE_THRESHOLD = 0.3f
 private const val MAX_DEGREE = 30f
 
+/**
+ * The activity that allow the user to swipe (like or skip) other users
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/match/MatchActivity.kt:24]:
```suggestion
 * The activity that allows the user to swipe (like or skip) other users
```

File: app/src/main/java/ch/epfl/sdp/blindly/match/MatchActivity.kt [hunk 13]
```diff
@@ -52,13 +55,19 @@ class MatchActivity : AppCompatActivity(), CardStackListener {
         }
     }
 
+    /**
+     * Do some ation when the card is dragged (still holded)
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/match/MatchActivity.kt:59]:
```suggestion
     * Do some action when the card is dragged (still holded)
```

File: app/src/main/java/ch/epfl/sdp/blindly/match/MatchActivity.kt (hunk 8 above)
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/match/MatchActivity.kt:91]:
```suggestion
     * Do some action when the card disappears
```

File: app/src/main/java/ch/epfl/sdp/blindly/match/MatchActivity.kt [hunk 14]
```diff
@@ -69,37 +78,57 @@ class MatchActivity : AppCompatActivity(), CardStackListener {
...
+     */
     override fun onCardDisappeared(view: View, position: Int) {
     }
 
-    private fun setupCardStackView() {
-        initialize()
-    }
-
     /**
      * Sets the 3 buttons up (like, rewind, skip)
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/match/MatchActivity.kt:100]:
```suggestion
     * Setup the 3 buttons (like, rewind, skip)
```

File: app/src/main/java/ch/epfl/sdp/blindly/match/Profile.kt [hunk 15]
```diff
@@ -37,7 +37,7 @@ data class Profile(
     /**
      * Describe the content as in Integer
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/match/Profile.kt:38]:
```suggestion
     * Describe the content as an Integer
```

File: app/src/main/java/ch/epfl/sdp/blindly/match/Profile.kt (hunk 9 above)
Comment (Lollipoke) [reply app/src/main/java/ch/epfl/sdp/blindly/match/Profile.kt:40]: I agree

**Review by Lollipoke — APPROVED** [review]

File: app/src/main/java/ch/epfl/sdp/blindly/match/Profile.kt (hunk 9 above)
Comment (r-ismer) [reply app/src/main/java/ch/epfl/sdp/blindly/match/Profile.kt:40]: return 1 if the object contains a content descriptor otherwise return 0

**Pull request merged**